src/main/java/com/starrocks/connector/flink/table/sink/StarRocksDynamicSinkFunctionV2.java

```diff
@@ -195,36 +180,67 @@ public void snapshotState(FunctionSnapshotContext functionSnapshotContext) throw
         StreamLoadSnapshot snapshot = sinkManager.snapshot();
 
         if (sinkManager.prepare(snapshot)) {
-            snapshotMap.put(functionSnapshotContext.getCheckpointId(), snapshot);
+            snapshotMap.put(functionSnapshotContext.getCheckpointId(), Collections.singletonList(snapshot));
 
             snapshotStates.clear();
-            snapshotStates.add(Tuple2.of(functionSnapshotContext.getCheckpointId(), snapshot));
+            snapshotStates.add(snapshotMap);
         } else {
             throw new RuntimeException("Snapshot state failed by prepare");
         }
     }
 
     @Override
     public void initializeState(FunctionInitializationContext functionInitializationContext) throws Exception {
-        ListStateDescriptor<Tuple2<Long, StreamLoadSnapshot>> descriptor =
+        ListStateDescriptor<Map<Long, List<StreamLoadSnapshot>>> descriptor =
```

Comment:
Can we encapsulate the state in a class like `Snapshot`, and define a serializer supports state compatibility (for example, write a version before serializer the snapshot data)? We can't always rely on state name to ensure compatibility (>﹏<).